wrapping/python/openmeeg/tests/test_python.py

```diff
@@ -19,39 +19,147 @@
+            if not os.path.exists(mesh_file):
+                print("Could not find mesh : " + mesh_file)
+            continue
+
+        if (len(words) > 1) and words[0].startswith("Interface"):
+            mesh_file = words[-1][1:-1]
+            if mesh_file.endswith(".tri"):
+                mesh_files.append(mesh_file)
+            if not os.path.exists(mesh_file):
+                print("Could not find mesh : " + mesh_file)
+            continue
+
+    for k, fname in enumerate(mesh_files):
+        if not os.path.isabs(fname):
+            mesh_files[k] = os.path.join(os.path.dirname(geom_file), fname)
+
+    print("Found : %s" % mesh_files)
+    return mesh_files
+
+
+def read_tri(fname):
+    """Read .tri file
+
+    Parameters
+    ----------
+    fname : str
+        The file to read.
+
+    Returns
+    -------
+    points : ndarray, shape (n_points, 3)
+        The vertices
+    normals : ndarray, shape (n_points, 3)
+        The normals at the vertices
+    faces : ndarray, shape (n_faces, 3)
+        The faces
+    """
+    assert fname.endswith(".tri")
+    fid = open(fname, "r")
+    # read the number of vertices
+    npoints = int(fid.readline().split()[1])
+
+    points = []
+    faces = []
+    normals = []
+
+    # fills the vertices arrays
+    for _ in range(npoints):
+        vals = fid.readline().split()
+        points.append(vals[:3])
+        normals.append(vals[3:])
+
+    # Read the number of triangles
+    n_faces = int(fid.readline().split()[1])
+    # create the list of triangles
+    for _ in range(n_faces):
+        vals = fid.readline().split()
+        faces.append(vals[:3])
+
+    # Convert to numpy arrays
+    points = np.array(points, np.float64)
+    normals = np.array(normals, np.float64)
+    faces = np.array(faces, np.int64)
+    return points, normals, faces
+
+
+@pytest.mark.parametrize(
+    "subject",
+    [
+        "Head1",
+        pytest.param("mne_sample_ico3", marks=pytest.mark.slow),
+    ],
+)
+@pytest.mark.parametrize("load_from_numpy", [True, False])
+def test_python(subject, data_path, load_from_numpy, tmp_path):
     # Load data
-    subject = "Head1"
     cond_file = op.join(data_path, subject, subject + ".cond")
     geom_file = op.join(data_path, subject, subject + ".geom")
     source_mesh_file = op.join(data_path, subject, subject + ".tri")
     dipole_file = op.join(data_path, subject, subject + ".dip")
     squidsFile = op.join(data_path, subject, subject + ".squids")
     patches_file = op.join(data_path, subject, subject + ".patches")
 
-    geom = om.Geometry(geom_file, cond_file)
-    assert geom.is_nested()
+    if load_from_numpy:
+        mesh_files = read_geom(geom_file)
+        meshes = []
+        for fname in mesh_files:  # we assume the order is form inner to outer
+            points, _, tris = read_tri(fname)
+            meshes.append((points, tris))
+        geom = om.make_nested_geometry(meshes, conductivity=(1, 0.0125, 1))
```

Comment:
Notice that what gets passed in the `load_from_numpy` case is `geom` created by `make_nested_geometry`, which takes a list of tuple of ndarray `(points, tris)`, *not* OpenMEEG `Mesh` objects.

Reply:
OK.